blob/service.go

```diff
@@ -165,11 +169,19 @@ func (s *Service) Included(
 	// but we have to guarantee that all our stored subtree roots will be on the same height(e.g. one
 	// level above shares).
 	// TODO(@vgonkivs): rework the implementation to perform all verification without network requests.
-	_, _, err := s.getByCommitment(ctx, height, nID, com)
-	if err != nil {
+	_, resProof, err := s.getByCommitment(ctx, height, nID, com)
+	switch err {
+	case nil:
+	case ErrBlobNotFound:
+		return false, nil
+	default:
 		return false, err
 	}
-	return true, nil
+
+	if !reflect.DeepEqual(proof, resProof) {
```

Comment:
There is a cleaner way to do it. Proof has getters for every private field, and we can easily make Equals method on the proof that accesses all of them and compares.

Reply:
But. it will be the same. This approach gives us less code

Reply:
It's effectively the same - yes, but reflect is a dirty hack for when we don't have access to the internal fields. Ideally, we would contribute to NMT with the Equals method. 

Also, note that this is going to be a hot path for a while(until v1), and having reflect on a hot path has its performance implications